infrastructure/devicepathresolver/scsi.go

```diff
@@ -0,0 +1,37 @@
+package devicepathresolver
+
+import (
+	boshsettings "github.com/cloudfoundry/bosh-agent/settings"
+	bosherr "github.com/cloudfoundry/bosh-utils/errors"
+)
+
+type scsi struct {
+	scsiVolumeID 	DevicePathResolver
+	scsiID          DevicePathResolver
+}
+
+func NewScsi(
+	scsiVolumeID DevicePathResolver,
+	scsiID DevicePathResolver,
+) DevicePathResolver {
+	return scsi{
+		scsiVolumeID: scsiVolumeID,
+		scsiID: scsiID,
+	}
+}
+
+func (scsiResolver scsi) GetRealDevicePath(diskSettings boshsettings.DiskSettings) (string, bool, error) {
+	var realPath string
+	var timeout bool
+	var err error
+
+	if len(diskSettings.ID) > 0 {
```

Comment:
Since we always setting ID for persistent disk in disk settings, that means we always going to resolve path by ID.